src/App.php

```diff
@@ -320,12 +330,18 @@ public function outputDebug($str)
      * @param string $output
      *
      * @throws \atk4\core\Exception
+     * @throws ExitApplicationException
      */
     public function terminate($output = null)
     {
         if ($output !== null) {
-            echo $output;
+            if ($this->isJsonRequest()) {
+                $this->outputResponseJSON($output);
+            } else {
```

Comment:
any response from atk4 is text/html, i made 2 function to respond with JSON or HTML headers
i have also made a outputResponse with custom headers for any special needs (SEE?)